roles/benchmarking_run_nvidiadl_ssd/tasks/main.yml

```diff
@@ -77,3 +77,21 @@
       oc describe pod/{{ benchmarking_nvidiadl_ssd_name }} -n {{ benchmarking_namespace }}
         > "{{ artifact_extra_logs_dir }}/pod_run-nvidiadl-ssd.descr"
     failed_when: false
+
+- name: Get average sample rate
+  shell:
+    tail -n9 "{{ artifact_extra_logs_dir }}/pod_run-nvidiadl-ssd.log" | head -n1 | awk '{ print $NF }'
+  register: samplerate
+
+- name: Fail if sample rate is below expectations
+  when: samplerate.stdout | float < 270.0
+  fail: msg="The average samples processed per second was unexpectedly low"
```

Comment:
(I know it's WIP), can you make this `270.0` a role parameter?
and I'm not actually sure where we want to do this speed check, 
here, or in the calling script

usually, I like to have the role general purpose, and let the calling script do the "error" checking:
270.0 value is directly linked with the INSTANCE_TYPE you chose, and maybe the threshold/epoch values